build.gradle

```diff
@@ -229,6 +229,22 @@ tasks.register("verifyVersions") {
   }
 }
 
+def generateUpgradeCompatibilityFile = tasks.register("generateUpgradeCompatibilityFile") {
+  def outputFile = project.layout.buildDirectory.file("rolling-upgrade-compatible.json")
+  def rollingUpgradeCompatibleVersions = buildParams.bwcVersions.wireCompatible - VersionProperties.elasticsearchVersion
+  inputs.property("rollingUpgradeCompatibleVersions", rollingUpgradeCompatibleVersions)
+  outputs.file(outputFile)
+  doLast {
+    def versionsString = rollingUpgradeCompatibleVersions.collect { "\"${it.toString()}\"" }.join(', ')
+    outputFile.get().asFile.write("""{"rolling_upgrade_compatible_versions" : [${versionsString}]}""")
+  }
+}
+
+def upgradeCompatibilityZip = tasks.register("upgradeCompatibilityZip", Zip) {
```

Comment:
we release other non zip plain files already (e.g. the taco file https://github.com/elastic/infra/blob/4d9aaee586fcfef69c4e97f1e90f877ca4fae0f5/cd/release/release-manager/project-configs/master/elasticsearch.gradle#L83) I wonder if we can skip the zipping bit

Reply:
The `.taco` file though _is_ a zip file. Maybe we can hack this but it's not clear if release manager does any validation here to ensure those are proper archives.